mesheryctl/internal/cli/root/system/update.go

```diff
@@ -67,15 +67,77 @@ var updateCmd = &cobra.Command{
 
 		switch currCtx.Platform {
 		case "docker":
-			log.Printf("Fetching latest docker-compose file for channel: %s...\n", currCtx.Channel)
-			err = utils.DownloadDockerComposeFile(currCtx, true)
+			if !utils.SkipResetFlag {
+
+				log.Printf("Fetching latest docker-compose file for channel: %s...\n", currCtx.Channel)
+				err = utils.DownloadDockerComposeFile(currCtx, true)
+				if err != nil {
+					return errors.Wrap(err, "failed to fetch docker-compose file")
+				}
+
+				log.Info("Updating Meshery...")
+
+				err = utils.UpdateMesheryContainers()
+				if err != nil {
+					return errors.Wrap(err, utils.SystemError("failed to update Meshery containers"))
+				}
+
+				// applying operator manifest
+				kubeClient, err := meshkitkube.New([]byte(""))
+				if err != nil {
+					return err
+				}
+
+				running, err := utils.IsMesheryRunning(currCtx.Platform)
+				if err != nil {
+					return err
+				}
+
+				if !running {
+					err = utils.CreateManifestsFolder()
+
+					if err != nil {
+						return err
+					}
+				}
+
+				err = utils.DownloadOperatorManifest()
+
+				if err != nil {
+					return err
+				}
+
+				err = utils.ApplyOperatorManifest(kubeClient, true, false)
+
+				if err != nil {
+					return err
+				}
+			}
+
+			skipUpdateFlag = true
+			err = restart()
+
 			if err != nil {
-				return errors.Wrap(err, "failed to fetch docker-compose file")
+				return err
 			}
 
-			err = utils.UpdateMesheryContainers()
+			utils.ViperDocker.SetConfigFile(utils.DefaultConfigPath)
+			err := utils.ViperDocker.ReadInConfig()
 			if err != nil {
-				return errors.Wrap(err, utils.SystemError("failed to update meshery containers"))
+				return err
+			}
+
+			dockerCompose := &config.MesheryCtlConfig{}
+			err = utils.ViperDocker.Unmarshal(&dockerCompose)
+			if err != nil {
+				return err
+			}
+
+			utils.ViperDocker.Set("contexts."+mctlCfg.CurrentContext, currCtx)
+
+			err = utils.ViperDocker.WriteConfig()
+			if err != nil {
+				return err
```

Comment:
What is the purpose of this block?

Reply:
This is to change the version of the current context to `latest` in `meshconfig`, since the version is effectively changed while updating.

Reply:
Okay, then why do we have multiple instances of viper for the same config file?

Reply:
I'll try to fix that @navendu-pottekkat 
Thanks!

Reply:
@navendu-pottekkat done this in my latest commit.

Reply:
Thank you. I will check it out.